### PR TITLE
Feature/upgrade deploy refactor

### PR DESCRIPTION
**Pull request by H34D**
## Description

Version 15 of the deployment script.

## Issues

 https://github.com/oceanprotocol/ocean/issues/289

### CODE REVIEW

File: scripts/deploy/contracts/artifacts/updateArtifact.js [hunk 1]
```diff
@@ -26,8 +27,12 @@ function updateArtifact(
 
     const artifactFileName = `${oldContractName}.${network.toLowerCase()}.json`
 
+    const resolvedArtifactsDir = path.resolve(artifactsDir)
     /* eslint-disable-next-line security/detect-non-literal-fs-filename */
-    const oldArtofactString = fs.readFileSync(`${artifactsDir}${artifactFileName}`, 'utf8').toString()
+    const oldArtofactString = fs.readFileSync(
```
Comment (aabdulwahed) [inline scripts/deploy/contracts/artifacts/updateArtifact.js]: `oldArtofactString` --> `oldArtifactString` small typo

File: scripts/deploy/contracts/deployContracts.js [hunk 2]
```diff
@@ -21,42 +25,31 @@ const NETWORK = process.env.NETWORK || 'development'
...
-]
+const contractNames = require('./contracts.json')
 
 async function deployContracts(
     web3,
     artifacts,
-    contracts,
+    contracts = [],
     forceWalletCreation = false,
+    deeperClean = false,
```
Comment (aabdulwahed) [inline scripts/deploy/contracts/deployContracts.js]: Can you please document what deeper clean does ? I don't understand what the difference between `clean` and `deep clean`

File: scripts/deploy/contracts/artifacts/updateArtifact.js (hunk 1 above)
Comment (H34D) [reply scripts/deploy/contracts/artifacts/updateArtifact.js]: solved, thanks for finding

File: scripts/deploy/contracts/deployContracts.js (hunk 2 above)
Comment (H34D) [reply scripts/deploy/contracts/deployContracts.js]: we have three:
- clean (will delete ./zos.json)
- clean deep (will delete ./zos.json and ./.zos.*)
- clean deeper (will delete ./zos.json, ./.zos.* and ./zos.dev-<the network id if the network you are currently connected to>.json)

**Pull request merged**